content/3_Silver/2P.mdx

```diff
@@ -31,6 +31,139 @@ frequency: 2
 	</Resource>
 </Resources>
 
+## Two Pointers
+
+The two pointers method iterates two pointers across an array, to track the start and end
+of an interval. It can also be used to track two values in order to satisfy a specific condition. 
+
+<FocusProblem problem="book" />
+
+## Solution - Books
+
+We want to find the longest continuous segment of books that can be read within $t$ minutes. 
+
+To accomplish this, we can define $\texttt{left}$ and $\texttt{right}$ as the start and end of the segment respectively.
+
+Since being able to read segment $\texttt{left} \dots \texttt{right}$ implies we can read segment $\texttt{left} \dots \texttt{right}-1$, we can use two-pointers (or, alternatively, [binary search](/silver/binary-search)).
+
+For every $\texttt{left}$ index, we need to increase our $\texttt{right}$ pointer until the total time for that segment exceeds $t$.
+
+We will keep track of the max segment size with $\texttt{ans}$, and take the maximum $\texttt{right} - \texttt{left}$ value.
+
+We can then remove the $\texttt{left}$-indexed book from $\texttt{cur}$, the total time of current segment, and increment $\texttt{left}$ by 1.
+
+**The $\texttt{right}$ pointer does not restart after an iteration, it simply keeps moving right.**
+
+<Info title="Pro-tip">
+
+You can visualize these pointers as maintaing a sliding window of books for this problem.
+
+</Info>
+
+### Implementation
+
+<LanguageSection>
+
+<CPPSection>
+
+
+```cpp
+#include <bits/stdc++.h> // see /general/running-code-locally
+using namespace std;
+
+using ll = long long;
+
+using vi = vector<int>;
+#define pb push_back
+#define all(x) begin(x), end(x)
+#define sz(x) (int)(x).size()
+
+using pi = pair<int,int>;
+#define f first
+#define s second
+#define mp make_pair
+
+void setIO(string name = "") {
+	cin.tie(0)->sync_with_stdio(0); // see /general/fast-io
+	if (sz(name)) {
+		freopen((name+".in").c_str(), "r", stdin); // see /general/io
+		freopen((name+".out").c_str(), "w", stdout);
+	}
+}
+
+
+int main() {
+	setIO();
```

Comment:
since this is the focus problem, I think it would be beneficial to comment the code.